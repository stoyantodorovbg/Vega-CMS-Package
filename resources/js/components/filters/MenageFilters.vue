<template>
    <section class="section-filters">
        <div class="row">
            <!-- Filter Options -->
            <div class="col-lg-12">
                <div class="row">
                    <div class="col-lg-12">
                        <button @click="changeEditing" type="button" class="btn btn-main">{{ getButtonValue() }}</button>
                    </div>
                    <div class="col-lg-12">
                        <div v-if="editing" class="form-group custom-filter-panel">
                            <div class="form-check"
                                 v-for="filter in filters">
                                <b-form-checkbox
                                        class="form-check-input"
                                        :checked="filter.visibility"
                                        @change="checkVisibility(filter.name, filter.visibility)">

                                        <div class="slot">
                                            {{ filter.name }}
                                        </div>
                                </b-form-checkbox>
                            </div>
                        </div>
                    </div>
                </div>
            </div>

            <!-- Filter Form -->
            <div class="col-lg-12">
                <filters :filters="filters"></filters>
            </div>
        </div>
    </section>

</template>

<script>
    import Filters from "./Filters";

    export default {
        components: { Filters },

        name: 'MenageFilters',

        props: {
            fields: {
                type: Array,
                default: function() {
                    return [
                        {
                            name: 'id',
                            visibility: true
                        }
                    ]
                },
            },
        },
       data() {
         return {
           editing: false,
         }
       },
      computed: {
            filters: {
                get: function() {
                     let filters = [];

                      this.fields.forEach((field) => {
                            filters.push({
                              name: field.name,
                              visibility: true
                            });
                      });

                    return filters;
                },
            },
        },
        methods: {
            checkVisibility(fieldName, fieldVisibility) {
                fieldVisibility = !fieldVisibility;
                this.changeFilterVisibility(fieldName, fieldVisibility);
            },
            getButtonValue() {
                return this.editing ? 'Menage filters - hide menu' : 'Menage filters - show menu';
            },
            changeEditing() {
                this.editing = !this.editing;
            },
            changeFilterVisibility(fieldName, fieldVisibility) {
                this.filters.map((obj) => {
                    if(obj.name === fieldName){
                        obj.visibility = fieldVisibility;
                    }
                })
            }
        }
    }
</script>
