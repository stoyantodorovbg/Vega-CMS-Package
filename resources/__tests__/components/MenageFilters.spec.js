import { mount } from '@vue/test-utils';
import MenageFilters from "../../js/components/filters/MenageFilters";

describe("ManageFilters Component", () => {
  test('should mount without crashing', () => {
      const wrapper = mount(MenageFilters);
      expect(wrapper).toMatchSnapshot();
  });
});
