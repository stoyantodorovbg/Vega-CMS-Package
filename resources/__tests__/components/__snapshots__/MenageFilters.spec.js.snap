// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ManageFilters Component should mount without crashing 1`] = `
VueWrapper {
  "_emitted": Object {},
  "_emittedByOrder": Array [],
  "isFunctionalComponent": undefined,
}
`;
